htdocs/compta/stats/casoc.php

```diff
@@ -458,8 +467,18 @@
 if ($subcat) {
 	print ' checked';
 }
-print'></td>';
-print '<td colspan="7" class="right">';
+
+print'>';
+// type filter (produit/service)
+print ' ';
+print '&nbsp;&nbsp;';
+print $langs->trans("Type").': ';
+$formcompany = new FormCompany($db);
+$sortparam = (empty($conf->global->SOCIETE_SORT_ON_TYPEENT) ? 'ASC' : $conf->global->SOCIETE_SORT_ON_TYPEENT); // NONE means we keep sort of original array, so we sort on position. ASC, means next function will sort on label.
```

Comment:
Use of $conf->global-> is forbidden. Use getDolGlobal...

Reply:
sorry, i have made this ugrade for a customer on 14.0.5 ...